File: CodeHelper/CodeHelper.Core/Parse/ParseResults/Workflows/Statements/SwitchStatement.cs
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

using CodeHelper.Core.Parse.ParseResults;
using CodeHelper.Core.Parse.ParseResults.Workflows.Expressions;
using System.Collections.Generic;

namespace CodeHelper.Core.Parse.ParseResults.Workflows.Statements
{
    /**
     *
     * @author Administrator
     */
    public class SwitchStatement : TokenPair
    {
        public Expression Key { get; set; }
        public List<CaseExpression> CaseExpressions { get; set; }

        public SwitchStatementInfo Parse()
        {
            var rslt = new SwitchStatementInfo(this);
            rslt.Key = Key.Parse();
            foreach (var c in this.CaseExpressions)
                rslt.CaseExpressions.Add(c.Parse());

            return rslt;
        }
    }
}

File: ToolBag/ParseTool/src/parsetool/workflow/models/statements/SwitchStatement.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package parsetool.workflow.models.statements;

import java.util.ArrayList;
import java.util.List;
import parsetool.models.common.TokenPair;
import parsetool.workflow.models.expressions.CaseExpression;
import parsetool.workflow.models.expressions.Expression;

/**
 *
 * @author Administrator
 */
public class SwitchStatement extends TokenPair {
    private Expression key;
    private List<CaseExpression> caseExpressions = new ArrayList<CaseExpression>();

    /**
     * @return the key
     */
    public Expression getKey() {
        return key;
    }

    /**
     * @param key the key to set
     */
    public void setKey(Expression key) {
        this.key = key;
    }

    /**
     * @return the caseExpressions
     */
    public List<CaseExpression> getCaseExpressions() {
        return caseExpressions;
    }

    /**
     * @param caseExpressions the caseExpressions to set
     */
    public void setCaseExpressions(List<CaseExpression> caseExpressions) {
        this.caseExpressions = caseExpressions;
    }
}
